### PR TITLE
support conditional expression _ ? _ : _

### PR DESCRIPTION
**Pull request by guanqun**
Most of the functionalities should be there, I'm thinking about possible corner cases.

I added several new test cases, they can all pass.

### CODE REVIEW

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 1]
```diff
@@ -86,6 +86,66 @@ BOOST_AUTO_TEST_CASE(exp_operator_const_signed)
...
+
+/*
+BOOST_AUTO_TEST_CASE(conditional_expression_as_left_value)
+{
+	char const* sourceCode = R"(
+		contract test {
+			function f(uint x) returns(uint d) {
+				uint y = 1;
+				uint z = 1;
+				(x > 10 ? y : z) = 3;
```
Comment (guanqun) [inline test/libsolidity/SolidityEndToEndTest.cpp:139]: I disabled this test case, because currently we don't support

```
(x) = 4;
```

Should we allow this?

**Comment by guanqun** [comment]
Take your time to review it, no hurries. :)

**Comment by VoR0220** [comment]
May want to try rerunning this through Jenkins...I can't find any details on a failed test.

EDIT: Upon further review...it appears to be a problem with memory allocation in Whisper...not sure how this might have caused that to fail. Ideas?

**Comment by VoR0220** [comment]
you want to rerun this. I ran a test case PR just for the sake of checking to make sure this would properly work and it went through. Try opening another PR.

**Comment by guanqun** [comment]
@VoR0220 Yes, it's a bit odd, I believe it's a build environment issue...

**Comment by guanqun** [comment]
Let me try another PR.
